src/resources/MachineLearning/RGAConfiguration/RGAConfigurationInterface.ts

```diff
@@ -0,0 +1,54 @@
+import {FilterConditions} from '../FilterConditions.js';
+import {DocumentRequirementStatus} from '../DocumentInterfaces.js';
+
+export interface RGAPreviewParams {
+    /**
+     * The sources to consider.
+     */
+    sources?: string[];
```

Comment:
This one is not optional 🙃

Reply:
Sorry for the confusion. The three properties should be used like this:
` (sources AND filterConditions) OR advancedQuery`. Advanced query includes both sources and any filter conditions. For example, this is accepted for both preview endpoints:

<img width="1462" alt="Screenshot 2023-11-21 at 4 05 42 PM" src="https://github.com/coveo/platform-client/assets/113382897/c935c77c-3052-4eaf-9c86-df8a6f8cbcb9">

Reply:
I checked also Semantic and it seems it is correct too

Reply:
I've tried it via swagger and I'm able to call it with only sources. From your comment I get that it should be [something like that](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBASwHY2FAZgQwMbDgJQHEBBABSmADcFgB3UzKTAWwGdiATSzJXDuAN4AoOKLisIAVyi5WAfgBccJFTQBuEWPQIANqigBhCEg4IYCY-KUrK6zaID0AKif3RTuABUAFngCOkmgAnnAw3pjwHMD6zMjArKG+cBwQ2JLMwCgJMBBwoDBM2DAAdHAGPEgQ8ABGeJKswPy0Zt5wVb5QyanpmfD5hebGcGCMLNForAA0ecUA5qUS0rLT2npocNjGpoNIU3kw2MVucK5i7nAAAgCiICxgOngXizLAALyvABQARACyIc+4L4ASjgxAAcgARS5gNDMHi9BD8cFQi46HizSSYWZvL5XJCzHQIVjeL4aM4nBxuTBcHh8ACKgSgQSUrAKyFmGgAvkIhKBILBECg0FhcAQSOQqDR6KM2ABlKQvBLCM7OU5nDw+PAA+KhXKbXaItBHclOSlnbVWcRs-EAbQAumSxKrjh5iEg4IwmCEIOg4Kt9OyNlszBZdsb1WatLp9EYTCHLIoPUggvbHaJqdxeI0GcFEzY7NzeeBoPAYEEYWKyBRqHQGEw2HBXpWJTXpfX2DSs-wAD7N6tSussVjypbxDRCfWsj1KIhVyW1mUJJsCDO07OM5lwL6+HQ6CBfTlwBwOOC0aAAa1YE8sNRn4v7C-bjcEFqUNu3wF3+7th+Pp4vV6TvA2B3nOraDg2y6vnA747nuXx2is0ZoLG2yhqwb4-keJ5nlAl48teux9FAUDEKBLYDouz4CNBsGfvBiEep29IbkoXyYBxB7YXkJHQNMmAJHywBFI0hFTmgUAAELkQ+bZDtRq5djmTJsRxmBfEhayGMGOwYTBWHcRJfEeoJ4DCagHBCEAA)

```
export interface RGAPreviewParamsAdvanced {
    sources?: never;
    filterConditions?: never;
    /**
     * The query that determines the documents to extract. Cannot be used with other document extraction parameters, e.g. sources, filter conditions, etc.
     *
     * @Example @source==("My source") AND @permanentid AND @language="English";
     */
    advancedQuery: string;
}

export interface RGAPreviewParamsSources {
    /**
     * The sources to consider.
     */
    sources?: string[];
    /**
     * An array of filtering conditions.
     */
    filterConditions?: FilterConditions[];
    advancedQuery?: never;
}

export type RGAPreviewParams = RGAPreviewParamsAdvanced | RGAPreviewParamsSources;

```

Reply:
I refactored both Semantic and RG!